envi/codeflow.py

```diff
@@ -348,6 +372,15 @@ def addEntryPoint(self, va, arch=envi.ARCH_DEFAULT):
 
         return va
 
+        
+    def _shouldDescend(self, va, arch):
+        # Check if this is already a known function.
+        if self._funcs.get(va) is not None:
```

Comment:
no, this is specific for halting at boundaries.  i just took the existing ` if self._funcs.get(va) is not None:` and moved it into this function call, instead of having spider-/cascading-logic in the codeflow calling code.

creating this function also allows us to implement the actual stopping code in envi.CodeFlow, while allowing the VivCodeFlow to override it for checking other things that aren't available outside of Vivisect (ie. we can't do them here, like Locations, XRefs, Exports, etc...).